Explorer/Assets/DCL/Audio/WorldAudioPlaybackController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using DCL.Diagnostics;
+using DCL.Optimization.Pools;
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Mathematics;
+using UnityEngine;
+using Object = UnityEngine.Object;
+using Random = UnityEngine.Random;
+
+namespace DCL.Audio
+{
+    public class WorldAudioPlaybackController : MonoBehaviour, IDisposable
+    {
+        [SerializeField]
+        private WorldAudioSettings audioSettings;
+        [SerializeField]
+        private AudioSource audioSourcePrefab;
+
+        private readonly Dictionary<int, List<AudioSource>> landscapeAudioSourcesPerIndex = new ();
+        private readonly Dictionary<int, List<AudioSource>> oceanAudioSourcesPerIndex = new ();
+        private GameObjectPool<AudioSource> audioSourcePool;
```

Comment:
Can you follow the nullable notation? You will see that `Dispose` can throw in this case and you should check that field there

Reply:
👍